tests/tests/test_state_scripts.py

```diff
@@ -632,6 +764,139 @@ class TestStateScripts(MenderTesting):
+        script_error_content = script_content + "exit 1"
+        broken_image = test_set.get("Rollback", False)
+
+        # Put artifact-scripts in the artifact.
+        artifact_script_dir = os.path.join(work_dir, "artifact-scripts")
+
+        if os.path.exists(work_dir):
+            shutil.rmtree(work_dir, ignore_errors=True)
+
+        os.mkdir(work_dir)
+        os.mkdir(artifact_script_dir)
+
+        new_rootfs = os.path.join(work_dir, "rootfs.ext4")
+        shutil.copy(conftest.get_valid_image(), new_rootfs)
+
+        ps = subprocess.Popen(
+            ["debugfs", "-w", new_rootfs], stdin=subprocess.PIPE)
+        ps.stdin.write("cd /etc/mender\n" "mkdir scripts\n" "cd scripts\n")
+        ps.stdin.close()
+        ps.wait()
+
+        for script in test_set.get("ScriptOrder"):
+            if not script.startswith("Artifact"):
+                # Not an artifact script, skip this one.
+                continue
+            with open(os.path.join(artifact_script_dir, script), "w") as fd:
+                if script in test_set.get("RebootScripts", []):
+                    fd.write(script_failure_content)
+                if script in test_set.get("RebootOnceScripts", []):
+                    fd.write(script_reboot_once)
+                elif script in test_set.get("StopSecondPartition", []):
+                    fd.write(script_stop_mender_on_entry)
+                elif script in test_set.get("ErrorScripts", []):
+                    fd.write(script_error_content)
+                else:
+                    fd.write(script_content)
+
+        # Now create the artifact, and make the deployment.
+        device_id = Helpers.ip_to_device_id_map([client])[client]
+        deployment_id = common_update_procedure(
+            install_image=new_rootfs,
+            broken_image=broken_image,
+            verify_status=True,
+            devices=[device_id],
+            scripts=[artifact_script_dir])[0]
+
+        try:
+
+            orig_part = Helpers.get_active_partition()
+
+            token = Helpers.place_reboot_token()
+
+            # handle case where the client has not finished the update
+            # path on the committed partition, but new partition is installed,
+            # thus we will not get a valid entrypoint into the uncommitted parition(reboot_leave)
+            # and the client will thus reboot straight after starting, and u-boot will
+            # fall back to the committed partition
+            if test_set.get("DoubleReboot", False):
+                token.verify_device_double_reboot()
+            else:
+                token.verify_reboot_performed(sleeptime=3, ntimes=3)
+
+            # if scripts are killed on a script that is run on the second partition,
+            # we need to wait for a second reboot
+            if test_set.get("DeviceDieSecondPartition", False) and len(
```

Comment:
Hmm, is it possible to get rid of `DeviceDieSecondPartition`? I think you only test it here, so always together with `StopSecondPartition`, hence it is redundant.

Reply:
I figured you would say this, and I agree. Initially I left it in for readability, but this is better handled with one parameter and a name change.